plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java

```diff
@@ -1156,6 +1157,9 @@ private void verifyRetryMode(ConnectorSession session, RetryMode retryMode)
             if (isNonTransactionalInsert(session)) {
                 throw new TrinoException(NOT_SUPPORTED, "Query and task retries are incompatible with non-transactional inserts");
             }
+            if (jdbcClient.supportsMerge() && isNonTransactionalMerge(session)) {
+                throw new TrinoException(NOT_SUPPORTED, "Query and task retries are incompatible with non-transactional merge");
```

Comment:
This feels wrong. This method is called not only in merge context. And any merge constraints should only  be relevant to merge statements.